hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java

```diff
@@ -484,34 +484,23 @@ public RecordStore getExistingRecordStore(int partitionId, String mapName) {
     }
 
     @Override
-    public PartitionIdSet getOwnedPartitions() {
-        PartitionIdSet partitions = ownedPartitions.get();
-        if (partitions == null) {
-            do {
-                reloadOwnedPartitions();
-                partitions = ownedPartitions.get();
-            } while (partitions == null);
+    public PartitionIdSet getOrInitCachedMemberPartitions() {
+        PartitionIdSet ownedPartitionIdSet = ownedPartitions.get();
```

Comment:
No need for `AtomicReference` anymore, you can convert `ownedPartitions` to a `volatile` field.

Reply:
fixed.